tests/functional/test_credentials.py

```diff
@@ -828,3 +828,16 @@ def test_credential_process_returns_error(self):
             reg = r"(?s)^((?!b').)*$"
             with self.assertRaisesRegexp(CredentialRetrievalError, reg):
                 session.get_credentials()
+
+
+class TestInstanceMetadataFetcher(unittest.TestCase):
```

Comment:
Let's subclass from `BaseSessionTest` and use its session if we can. Mainly there are some environment variables that can be set to affect behavior of this provider and we should make sure they do not get inadvertently set.